core/shared/src/main/scala/scalaz/zio/duration/Duration.scala

```diff
@@ -20,6 +20,15 @@ sealed trait Duration extends Ordered[Duration] with Serializable with Product {
     case f: Duration.Finite => finite(f)
   }
 
+  /* Number of milliseconds. Negative values indicate infinity */
+  def toMillis: Long
```

Comment:
Having infinite duration, this doesn't seem sound to me. `Option[Long]`? If `Long` why `-1` instead of `Long.MaxValue` - both are wrong for me (and both in different way wrong).
I'm sorry for rising concerns without showing and proving that some other way is better (in every or just some cases).
I have some wild idea to consider: to remove `Infinite` duration and add `Max` and `Min` (I think is around 292 years with Long and nanos).
Such `Max` is totally fine for all practical Scheduling. 

Pros:
* no chance for irritating error when someone uses `toMillis` with 3rd party API and it turns out he send -1.
* sound algebra in more cases: `Max * Zero == Zero`, when `Infinity * Zero` is not well defined


Of course there are still decisions to make: I still don't know if Max + Max should return Max or should `+` return Option.

Reply:
I did this for two reasons:

a) `toNanos` and `toMillis` now represent the inverse of `apply`, which returns `Infinity` for negative integers
b) `DurationSyntax` now actually works much better because `10.millis.toNanos` is now a thing

It is true that, strictly speaking, `-1` is irritating. I fully agree with that.

To be fair, I am mostly interested in the syntax module here :)
And from my POV, it would be sufficient to allow negative durations. I am unsure why it was decided not to have those and I am not sure what the implications in the scheduling bits would be.

Reply:
Ad a) `Finite(-1)` returns `Infinity` - that is bad IMO. I'll take care of it if I can. But I would go with John advice and return `Long.MaxValue` for `Inifinity.toMillis` and `.toNanos.

Mapping negatives to `Infinity` covers some of overflow problems but not all of them - but that is other discussion.

Reply:
How about this: Let's do the remapping of Infinity in an additional PR. In that, we can then also allow negative instances of `Finite`, which I find useful sometimes.

Reply:
I'm OK with this. However I don't have rights to approve.